mne/io/pick.py

```diff
@@ -773,12 +777,129 @@ def _pick_aux_channels(info, exclude='bads'):
                       ref_meg=False, exclude=exclude)
 
 
-def _pick_data_or_ica(info):
+def _pick_data_or_ica(info, exclude=()):
     """Pick only data or ICA channels."""
     if any(ch_name.startswith('ICA') for ch_name in info['ch_names']):
-        picks = pick_types(info, exclude=[], misc=True)
+        picks = pick_types(info, exclude=exclude, misc=True)
+    else:
+        picks = _pick_data_channels(info, exclude=exclude, with_ref_meg=True)
+    return picks
+
+
+def _picks_to_idx(info, picks, none='data', exclude='bads', allow_empty=False):
+    """Convert and check pick validity."""
+    #
+    # None -> all, data, or data_or_ica (ndarray of int)
```

Comment:
as it is a private function this comment can be put in the docstring

Reply:
oh, ignore that comment, I understand now that these are just titles for the code "chapters". :)

Reply:
maybe we should unify the code "chapters" at some point. 'cos we find them in different flavors in the codebase.

Reply:
but lets do that another day